src/components/global-setting/index.vue

```diff
@@ -27,46 +26,53 @@
   import { Message } from '@arco-design/web-vue'
   import { useI18n } from 'vue-i18n'
   import { useClipboard } from '@vueuse/core'
-  import { useAppStore } from '@/store'
+  import { useAppStore, useDataBaseStore } from '@/store'
+  import axios from 'axios'
   import Block from './block.vue'
 
   const emit = defineEmits(['cancel'])
 
   const appStore = useAppStore()
+  const dataBaseStore = useDataBaseStore()
+
   const { t } = useI18n()
   const { copy } = useClipboard()
   const visible = computed(() => appStore.globalSettings)
-  const contentOpts = computed(() => [
-    { name: 'settings.navbar', key: 'navbar', defaultVal: appStore.navbar },
-    {
-      name: 'settings.menu',
-      key: 'menu',
-      defaultVal: appStore.menu,
-    },
-    { name: 'settings.footer', key: 'footer', defaultVal: appStore.footer },
-    { name: 'settings.tabBar', key: 'tabBar', defaultVal: appStore.tabBar },
+  const authOpts = computed(() => [
+    { name: 'settings.username', key: 'principal', defaultVal: appStore.principal, type: 'input' },
     {
-      name: 'settings.menuFromServer',
-      key: 'menuFromServer',
-      defaultVal: appStore.menuFromServer,
+      name: 'settings.password',
+      key: 'credential',
+      defaultVal: appStore.credential,
+      type: 'password',
     },
+  ])
+  const databaseOpts = computed(() => [
     {
-      name: 'settings.menuWidth',
-      key: 'menuWidth',
-      defaultVal: appStore.menuWidth,
-      type: 'number',
+      name: 'settings.databaseURL',
+      key: 'databaseURL',
+      defaultVal: appStore.databaseURL,
+      type: 'input',
     },
-  ])
-  const othersOpts = computed(() => [
     {
-      name: 'settings.colorWeak',
-      key: 'colorWeak',
-      defaultVal: appStore.colorWeak,
+      name: 'settings.database',
+      key: 'database',
+      defaultVal: appStore.database,
+      selectOps: appStore.databaseList,
+      type: 'select',
+      disabled: appStore.isCloud,
     },
   ])
 
   const cancel = () => {
     appStore.updateSettings({ globalSettings: false })
+    axios.defaults.baseURL = appStore.databaseURL
+    if (appStore.codeType === 'sql') {
```

Comment:
Consider use constants for `'sql'` as we might rename this page

Reply:
Will fix this when we add PromQL